models/user.js

```diff
@@ -3,18 +3,34 @@ const bcrypt   = require('bcrypt-nodejs')
 const Schema = mongoose.Schema
 
 const UserSchema = new Schema({
-  username: String,
+  username: { type: String, unique: true, lowercase: true },
   password: String
 }, {timestamps: true})
 
 
-UserSchema.methods.generateHash = function(password) {
-    return bcrypt.hashSync(password, bcrypt.genSaltSync(8), null);
-};
+// On Save Hook, encrypt password
```

Comment:
This function will run automatically before saving a new user to the `users` collection.